core/partitioning/shape_analysis.cpp

```diff
@@ -222,29 +227,39 @@ void getSegmentsOutputByRunning(
 
   auto target_device = partitioning_info.getGPUDeviceString();
 
-  // auto int64 <=> int32 conversion
-  if (seg_block.target() == SegmentedBlock::kTorch && partitioning_info.truncate_long_and_double) {
+  // auto int64 <=> int32 conversion + int8 <=> int32 conversion for non-quantized models
+  if (seg_block.target() == SegmentedBlock::kTorch) {
     // First, check if there is Int64 input
-    for (size_t i = 0; i < seg_block.inputs().size(); ++i) {
-      if (ivalues_maps[seg_block.raw_inputs()[i]].isTensor()) {
-        auto cur_ivalue = ivalues_maps[seg_block.raw_inputs()[i]];
-        at::ScalarType t = cur_ivalue.toTensor().scalar_type();
-        if (t == at::kLong) {
-          // we add a cast operation to cast the type to Int64
-          auto cast_node = createCastNode(seg_block, i, true, target_device);
-          seg_block.g()->prependNode(cast_node);
-          seg_block.inputs()[i]->replaceAllUsesAfterNodeWith(cast_node, cast_node->outputs()[0]);
+    if (partitioning_info.truncate_long_and_double) {
+      for (size_t i = 0; i < seg_block.inputs().size(); ++i) {
+        if (ivalues_maps[seg_block.raw_inputs()[i]].isTensor()) {
+          auto cur_ivalue = ivalues_maps[seg_block.raw_inputs()[i]];
+          at::ScalarType t = cur_ivalue.toTensor().scalar_type();
+          if (t == at::kLong) {
+            // we add a cast operation to cast the type to Int64
+            auto cast_node = createCastNode(seg_block, i, true, target_device);
+            seg_block.g()->prependNode(cast_node);
+            seg_block.inputs()[i]->replaceAllUsesAfterNodeWith(cast_node, cast_node->outputs()[0]);
+          }
         }
       }
     }
+
     for (size_t i = 0; i < seg_block.outputs().size(); ++i) {
       if (ivalues_maps[seg_block.raw_outputs()[i]].isTensor()) {
         auto cur_ivalue = ivalues_maps[seg_block.raw_outputs()[i]];
         at::ScalarType t = cur_ivalue.toTensor().scalar_type();
-        if (t == at::kLong) {
+
+        // If the input has type Long and truncation was requested, insert truncate
+        if (t == at::kLong && partitioning_info.truncate_long_and_double) {
           auto cast_node = createCastNode(seg_block, i, false, target_device);
           seg_block.g()->appendNode(cast_node);
           seg_block.g()->block()->replaceOutput(i, cast_node->outputs()[0]);
+        } else if (t == at::kByte && partitioning_info.cast_int8_inputs) {
+          // If the input has type Byte and truncation was requested, insert Integer cast
+          auto cast_node = createCastNode(seg_block, i, false, target_device, /*force_create_node=*/true);
+          seg_block.g()->appendNode(cast_node);
+          seg_block.g()->block()->replaceOutput(i, cast_node->outputs()[0]);
```

Comment:
Can we add LOG_DEBUG statements to ensure we log all the casting and user understands the flow by the logs ?